lib/db/index.js

```diff
@@ -19,30 +19,36 @@ function connect() {
   }
 
   return srv.resolve(options.url)
-    .then(function(records) {
-      function next() {
-        var record = records.shift()
-
-        if (!record) {
-          throw new Error('No hosts left to try')
-        }
-
-        log.info('Connecting to %s:%d', record.name, record.port)
-
-        return r.connect({
-            host: record.name
-          , port: record.port
-          , db: options.db
-          , authKey: options.authKey
-          })
-          .catch(r.Error.RqlDriverError, function() {
-            log.info('Unable to connect to %s:%d', record.name, record.port)
-            return next()
-          })
+  .then(function(records) {
+    function next() {
+      var record = records.shift()
+
+      if (!record) {
+        throw new Error('No hosts left to try')
       }
 
-      return next()
-    })
+      log.info('Connecting to %s:%d', record.name, record.port)
+
+      return r.connect({
+          host: record.name
+        , port: record.port
+        , db: options.db
+        , authKey: options.authKey
+        })
+        .catch(r.Error.RqlDriverError, function() {
+          log.info('Unable to connect to %s:%d', record.name, record.port)
+          return next()
+        })
+    }
+
+    // DB Connection Recovery
+    return next()
```

Comment:
how about limitations in retries?

Reply:
Sorry if I didn't understand, should it try to reconnect a specific number of times? In that case, how many times?